### PR TITLE
Add aligned allocator for ListPoint2D

### PR DESCRIPTION
**Pull request by Phil26AT**
Problem: COLMAP uses [Eigens aligned allocator](https://eigen.tuxfamily.org/dox/group__TopicStlContainers.html) for vectors of [Point2D](https://github.com/colmap/colmap/blob/dev/src/base/point2d.h#L96).

This PR ensures that `pycolmap.ListPoint2D` also uses this aligned allocator (previously caused `MaxRecursionError` in some cases).

### CODE REVIEW

File: reconstruction/point2D.cc [hunk 1]
```diff
@@ -1,5 +1,8 @@
 // Author: Philipp Lindenberger (Phil26AT)
 
+// Use Eigens aligned allocator for vectors
+#include<Eigen/StdVector>
```
Comment (Skydes) [inline reconstruction/point2D.cc:4]: Code style: extra space after the include?

**Comment by guohengkai** [comment]
@Phil26AT It seems that colmap has added a macro to deal with the alignment: https://github.com/colmap/colmap/blob/master/src/base/point2d.h#L96
And after I patched this commit, the error still occurs.

**Comment by Phil26AT** [comment]
@guohengkai I am a bit clueless then, for me this fixes the issue. Are you using ceres v2.2? Then maybe try to compile colmap explicitly with --std=c++17 (this is still missing in the colmap cmake, atm it will still use c++14, but ceres uses c++17).

Can you maybe open an issue with your ceres and colmap commit hash?

**Comment by guohengkai** [comment]
@Phil26AT Sure. I will construct a minimal code snipper to reproduce it and submit an issue later. Thanks.

**Comment by Skydes** [comment]
Hey @guohengkai @Phil26AT, any progress on this? Thanks!

**Review by mihaidusmanu — APPROVED** [review]

File: reconstruction/point2D.cc (hunk 1 above)
Comment (mihaidusmanu) [reply reconstruction/point2D.cc:4]: Will add clang-format.

**Pull request merged**